Parquet/src/main/java/io/deephaven/parquet/ParquetFileReader.java

```diff
@@ -75,6 +77,56 @@ public ParquetFileReader(String filePath, SeekableChannelsProvider channelsProvi
         type = fromParquetSchema(fileMetaData.schema,fileMetaData.column_orders);
     }
 
+    private HashSet<String> columnsWithDictionaryUsedOnEveryDataPage = null;
+    public HashSet<String> getColumnsWithDictionaryUsedOnEveryDataPage() {
+        if (columnsWithDictionaryUsedOnEveryDataPage == null) {
+            columnsWithDictionaryUsedOnEveryDataPage = calculateColumnsWithDictionaryUsedOnEveryDataPage();
+        }
+        return columnsWithDictionaryUsedOnEveryDataPage;
+    }
+
+    private HashSet<String> calculateColumnsWithDictionaryUsedOnEveryDataPage() {
+        final HashSet<String> result = new HashSet<>(fileMetaData.getSchemaSize());
+        int rowGroupSeq = 0;
+        for (RowGroup rowGroup : fileMetaData.getRow_groups()) {
+            // On the first pass, for row group zero, we are going to add all columns to the set
+            // that satisfy the restriction.
+            // On later passes after zero, we will remove any column that does not satisfy
+            // the restriction.
+            for (ColumnChunk columnChunk : rowGroup.columns) {
+                final ColumnMetaData columnMeta = columnChunk.getMeta_data();
+                final String parquetColumnName = columnMeta.path_in_schema.get(0);
+                if (rowGroupSeq > 0 && !result.contains(parquetColumnName)) {
+                    continue;
+                }
+                if (columnMeta.encoding_stats == null) {
```

Comment:
What does this mean? Is it possible to have stats on the first column chunk, but not on subsequent ones? Seems like we're assuming that if that happens, it doesn't create a conflict. I might rather assume that it does, and remove this column from the set if we see no stats.

Reply:
The logic was mudded.  Is easier to reason about if we just explicitly separate the first pass.